sys/net/gnrc/sock/ip/gnrc_sock_ip.c

```diff
@@ -156,6 +161,16 @@ ssize_t sock_ip_recv_buf_aux(sock_ip_t *sock, void **data, void **buf_ctx,
     if ((aux != NULL) && (aux->flags & SOCK_AUX_GET_LOCAL)) {
         aux->flags &= ~(SOCK_AUX_GET_LOCAL);
     }
+#endif
+#if IS_USED(MODULE_SOCK_AUX_TIMESTAMP)
+    if ((aux != NULL) && (aux->flags & SOCK_AUX_GET_TIMESTAMP)) {
+        /* check if network interface did provide a timestamp; this depends on
+         * hardware support. A timestamp of zero is used to indicate a missing
+         * timestamp */
+        if (aux->timestamp > 0) {
```

Comment:
What, for whatever reason the timestamp actually is 0?

Reply:
As this is in nanoseconds, the odds are good this never happens. The duration the clock is zero will likely not be 1 nanoseconds small, as the clock is unlikely to run at 1 GHz. But let's say the clock runs at least at 1 MHz, the issue only can occur within the first µs after the clock is started.

If this happens, the stack would communicate that no TX timestamp is available for the given frame. This is something the user space is to take into account anyway. E.g. there are some Ethernet cards (for desktop / server machines) which only can timestamp one frame at a time. So while time stamping an outgoing frame you cannot get a time stamp for any frame that is received simultaneously. As a result, most incoming frames will have a time stamp, but occasionally you'll get a frame without a timestamp.

The same issue could also easily apply to embedded hardware. (The STM32 Ethernet peripheral that I'm using has no such limitation - or at least I am not aware of it. But I would still carefully check for every received datagram / packet, if it indeed contains a timestamp just to be sure.)

Reply:
Mh, your arguments make sense, I would nevertheless find it more elegant, when the `get_timestamp()` function would return an error that could be checked here (maybe provided via the struct), and the flag to be cleared based on that.

Reply:
4 bytes for `tests/gnrc_sock_ip` and no change for `tests/gnrc_sock_udp`. I'll add it.